### PR TITLE
Move everything to subgraph

### PR DESCRIPTION
**Pull request by rhlsthrm**

### CODE REVIEW

File: packages/sdk/src/sdk.ts [hunk 1]
```diff
@@ -983,7 +939,15 @@ export class NxtpSdk {
     timeout?: number,
   ): void {
     const args = [timeout, callback].filter((x) => !!x);
-    this.evts[event].pipe(filter).attachOnce(...(args as [number, any]));
+    if (
+      [NxtpSdkEvents.SenderTransactionPrepared, NxtpSdkEvents.ReceiverTransactionPrepared].includes(
+        event as "SenderTransactionPrepared" | "ReceiverTransactionPrepared", // so stupid...
```
Comment (jakekidd) [inline packages/sdk/src/sdk.ts:944]: why is this needed? lol

Comment (rhlsthrm) [reply packages/sdk/src/sdk.ts:944]: Because the `NxtpSdkEvents` has more than just those lol.

File: packages/sdk/src/subgraph.ts [hunk 2]
```diff
@@ -114,14 +155,30 @@ export class Subgraph {
...
+          const tx: ActiveTransaction = {
+            txData: convertTransactionToTxData(txData),
+            bidSignature: txData.bidSignature,
+            caller: txData.prepareCaller,
+            encodedBid: txData.encodedBid,
+            encryptedCallData: txData.encryptedCallData,
+            status: SubgraphEvents.SenderTransactionPrepared,
+          };
+          // if it doesn't exist in the map, we need to post about it
+          // also if it is in the SenderPrepared status, update the status and post about it
```
Comment (LayneHaber) [inline packages/sdk/src/subgraph.ts:168]: Where is the status updated?

File: packages/sdk/src/subgraph.ts [hunk 3]
```diff
@@ -152,7 +217,115 @@ export class Subgraph {
...
+      .filter(([, tx]) => !allIds.includes(tx.txData.transactionId))
+      .map(([, tx]) => tx); // get just the array of txs from the map
+    const unknownPerChain: Record<number, ActiveTransaction[]> = {};
+    unknownTxs.forEach((tx) => {
+      if (!unknownPerChain[tx.txData.receivingChainId]) {
+        unknownPerChain[tx.txData.receivingChainId] = [tx];
+      } else {
+        unknownPerChain[tx.txData.receivingChainId].push(tx);
+      }
+    });
```
Comment (LayneHaber) [inline packages/sdk/src/subgraph.ts:235]: why does it look at the receiving chain id?

Comment (LayneHaber) [reply packages/sdk/src/subgraph.ts:235]: Definitely need more comments to highlight the overall thinking/process here. im still not exactly sure what this bit is doing

**Review by LayneHaber — CHANGES_REQUESTED** [review]

File: packages/sdk/src/sdk.ts [hunk 4]
```diff
@@ -964,7 +912,15 @@ export class NxtpSdk {
     timeout?: number,
   ): void {
     const args = [timeout, callback].filter((x) => !!x);
-    this.evts[event].pipe(filter).attach(...(args as [number, any]));
+    if (
+      [NxtpSdkEvents.SenderTransactionPrepared, NxtpSdkEvents.ReceiverTransactionPrepared].includes(
+        event as "SenderTransactionPrepared" | "ReceiverTransactionPrepared", // so stupid...
+      )
```
[the remark this answers is not in the document]
Comment (rhlsthrm) [reply packages/sdk/src/sdk.ts:918]: Yeah in fact maybe the SDK events _are_ the subgraph events.

[the remark this answers is not in the document]
Comment (rhlsthrm) [reply packages/sdk/src/sdk.ts:918]: No, there are more events in the SDK besides subgraph events so we can't just do that. Will see if I can do anything else.

File: packages/router/src/subgraph.ts [hunk 5]
```diff
@@ -145,10 +145,10 @@ export class Subgraph {
   private subgraphLoop() {
     const method = "startLoop";
```
[the remark this answers is not in the document]
Comment (jakekidd) [reply packages/router/src/subgraph.ts:146]: This is actually why I've been using the `.name` property, just to enforce this

You can get the same string by saying `this.subgraphLoop.name`

File: packages/sdk/src/sdk.ts [hunk 6]
```diff
@@ -896,56 +887,13 @@ export class NxtpSdk {
   /**
    * Sets up all listeners on the transaction manager across all chains, and for autocompletion of txs on the sdk
    */
-  public establishListeners(): void {
-    this.transactionManager.establishListeners();
-    this.setupListeners();
-  }
+  public establishListeners(): void {}
```
[the remark this answers is not in the document]
Comment (jakekidd) [reply packages/sdk/src/sdk.ts:890]: If we're going to keep it, I'd prefer at least a NotImplementedError being fired

File: packages/router/src/subgraph.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (LayneHaber) [reply packages/router/src/subgraph.ts:146]: You can forget to change the method you're referencing sometimes 🙃

**Review by LayneHaber — APPROVED** [review]
